homeassistant/components/emulated_roku/binding.py

```diff
@@ -0,0 +1,105 @@
+"""Bridge between emulated_roku and Home Assistant."""
+import logging
+
+from homeassistant.core import EventOrigin
+
+LOGGER = logging.getLogger('homeassistant.components.emulated_roku')
+
+EVENT_ROKU_COMMAND = 'roku_command'
+
+ATTR_COMMAND_TYPE = 'type'
+ATTR_SOURCE_NAME = 'source_name'
+ATTR_KEY = 'key'
+ATTR_APP_ID = 'app_id'
+
+ROKU_COMMAND_KEYDOWN = 'keydown'
+ROKU_COMMAND_KEYUP = 'keyup'
+ROKU_COMMAND_KEYPRESS = 'keypress'
+ROKU_COMMAND_LAUNCH = 'launch'
+
+
+class EmulatedRoku:
+    """Manages an emulated_roku server."""
+
+    def __init__(self, hass, name, host_ip, listen_port,
+                 advertise_ip, advertise_port, upnp_bind_multicast):
+        """Initialize the properties."""
+        self.hass = hass
+
+        self.roku_usn = name
+        self.host_ip = host_ip
+        self.listen_port = listen_port
+
+        self.advertise_port = advertise_port
+        self.advertise_ip = advertise_ip
+
+        self.bind_multicast = upnp_bind_multicast
+
+        self.api_server = None
+        self.ssdp_server = None
+
+    async def async_setup(self):
+        """Start the emulated_roku server."""
+        from emulated_roku import RokuCommandHandler, make_roku_api
+
+        class EventCommandHandler(RokuCommandHandler):
+            """emulated_roku command handler to turn commands into events."""
+
+            def __init__(self, hass):
+                self.hass = hass
+
+            def on_keydown(self, roku_usn, key):
+                """Handle keydown event."""
+                self.hass.bus.async_fire(EVENT_ROKU_COMMAND, {
+                    ATTR_SOURCE_NAME: roku_usn,
+                    ATTR_COMMAND_TYPE: ROKU_COMMAND_KEYDOWN,
+                    ATTR_KEY: key
+                }, EventOrigin.remote)
```

Comment:
Don't use remote origin. That's only for connected hass instances (which we don't support anymore)